asapdiscovery-ml/asapdiscovery/ml/inference/inference.py

```diff
@@ -173,8 +176,43 @@ def predict(self, g: dgl.DGLGraph):
         g : dgl.DGLGraph
             DGLGraph object.
 
+        Returns
+        -------
+        np.ndarray
+            Predictions for each graph.
         """
         with torch.no_grad():
             output_tensor = self.model(g, g.ndata["h"])
+            # unsure exactly why this reshape is done
             output_tensor = torch.reshape(output_tensor, (-1, 1))
-            return output_tensor.cpu().numpy()
+            # we ravel it away to get a 1D array
+            return output_tensor.cpu().numpy().ravel()
+
+    def predict_from_smiles(
+        self, smiles: Union[str, list[str]], **kwargs
+    ) -> Union[np.ndarray, float]:
+        """Predict on a list of SMILES strings, or a single SMILES string.
+
+        Parameters
+        ----------
+        smiles : Union[str, List[str]]
+            SMILES string or list of SMILES strings.
+
+        Returns
+        -------
+        np.ndarray or float
+            Predictions for each graph, or a single prediction if only one SMILES string is provided.
+        """
+        if isinstance(smiles, str):
+            smiles = [smiles]
+
+        gids = GraphInferenceDataset(
+            smiles, node_featurizer=CanonicalAtomFeaturizer(), **kwargs
+        )
+
+        data = [self.predict(g) for g in gids]
+        data = np.concatenate(np.asarray(data))
```

Comment:
is there a reason we need the `np.concatenate` here? I don't think it matters, I'm just curious

Reply:
good spotting, the `asarray` probably does the job.

Reply:
Actually wrong, the concatenate is needed to flatten.